docs/table-migration.md

```diff
@@ -0,0 +1,89 @@
+![CTAS Migration](../../../img/iceberg-CTAS.png)
+### In-Place Migration
+In-place migration retains the existing data files but adds Iceberg metadata on top of them. This approach is faster and does not require copying data, making it more suitable for production use cases.
+![In-Place Migration](../../../img/iceberg-In-place.png)
+## In-Place Migration Actions
+Apache Iceberg primarily supports the in-place migration approach, which includes three important actions:
+
+1. Snapshot Table
+2. Migrate Table
+3. Add Files
+
+### Snapshot Table
+The Snapshot Table action creates a new iceberg table with the same schema and partitioning as the source table, leaving the source table unchanged during and after the action.
+
+**Step 1:** Create a new Iceberg table with the same metadata (schema, partition spec, etc.) as the source table
+![Snapshot Table Step 1](../../../img/iceberg-snapshotaction-step1.png)
+
+**Step 2:** Commit all data files across all partitions to the new Iceberg table. The source table remains unchanged.
+![Snapshot Table Step 2](../../../img/iceberg-snapshotaction-step2.png)
+### Migrate Table
+The Migrate Table action also creates a new Iceberg table with the same schema and partitioning as the source table. However, during the action execution, it locks and drops the source table from the catalog.
+Consequently, Migrate Table requires all readers and writers working on the source table to be stopped before the action is performed.
+
+**Step 1:** Stop all readers and writers interacting with the source table
+![Migrate Table Step 1](../../../img/iceberg-migrateaction-step1.png)
+
+**Step 2:** Create a new Iceberg table with the same metadata (schema, partition spec, etc.) as the source table. Rename the source table for a backup in case of failure and rollback.
+![Migrate Table Step 2](../../../img/iceberg-migrateaction-step2.png)
+
+**Step 3:** Commit all data files across all partitions to the new Iceberg table. Drop the source table.
+![Migrate Table Step 3](../../../img/iceberg-migrateaction-step3.png)
+### Add Files
+After the initial step (either Snapshot Table or Migrate Table), it is common to find some data files that have not been migrated. These files often originate from concurrent writers who continue writing to the source table during or after the migration process.
+In practice, these files can be new data files in Hive tables or new snapshots (versions) of Delta Lake tables. The Add Files action is essential for incorporating these files into the Iceberg table.
+
+## In-Place Migration Completion
+Once all data files have been migrated and there are no more concurrent writers writing to the source table, the migration process is complete.
+Readers and writers can now switch to the new Iceberg table for their operations.
+
+## Migration Implementation: From Hive/Spark to Iceberg
+Apache Hive and Apache Spark are two popular data warehouse systems used for big data processing and analysis.
```

Comment:
we should remove opinions and evaluations of other systems in OSS docs. We just need an intro like Hive tables backed by Parquet, ORC and Avro file formats can be migrated to Iceberg.